test/functional/apps/dashboard/dashboard_error_handling.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { FtrProviderContext } from '../../ftr_provider_context';
+
+export default function ({ getService, getPageObjects }: FtrProviderContext) {
+  const esArchiver = getService('esArchiver');
+  const PageObjects = getPageObjects(['dashboard', 'header', 'common']);
+  const browser = getService('browser');
+  const testSubjects = getService('testSubjects');
+
+  /**
+   * Common test suite for testing exception scenarious within dashboard
+   */
+  describe('dashboard error handling', () => {
+    before(async () => {
+      await esArchiver.loadIfNeeded('dashboard/current/kibana');
+      await PageObjects.common.navigateToApp('dashboard');
+    });
+
+    it('recreate index pattern link works', async () => {
+      await PageObjects.dashboard.gotoDashboardLandingPage();
+      await PageObjects.dashboard.loadSavedDashboard('dashboard with missing index pattern');
+      await PageObjects.header.waitUntilLoadingHasFinished();
+      const errorEmbeddable = await testSubjects.find('embeddableStackError');
+      await (await errorEmbeddable.findByTagName('a')).click();
+      await browser.switchTab(1);
+      await testSubjects.existOrFail('createIndexPatternButton');
+      await browser.closeCurrentWindow();
+      await browser.switchTab(0);
```

Comment:
Thank you for adding the test. I have concerns about closing tab in `it` function:

`await testSubjects.existOrFail('createIndexPatternButton');` may fail and in this case the following tests will be run with 2 tabs.

I suggest adding counter `let tabsCount = 1;` in main function, increase it after `await browser.switchTab(1);` and move closing tab in `after`:

```
    after(async () => {
      if (tabsCount > 1) {
        await browser.closeCurrentWindow();
        await browser.switchTab(0);
      }
    });
```

Reply:
This makes sense! thank you!

Reply:
https://github.com/elastic/kibana/pull/67280/commits/2b6167fb1f798392ba2380de77647f56245bb02f